lib/checkTemplateMatch.js

```diff
@@ -0,0 +1,17 @@
+const res = await context.github.repos.getContent(this.repo({
+  path: '.github/ISSUE_TEMPLATE.md'
+}))
+
+const issueTemplate = Buffer.from(res.data.content, 'base64').toString()
+
+function checkTemplateMatch(body) {
+  const lines = issueTemplate.split("\n*")
+  lines.forEach(line => {
+    if (!body.includes(line)) {
```

Comment:
How does this work for templates with forms you have to fill out? For example:

```markdown
**Foo version**: [enter your version of Foo here]
**OS version**: [enter your operating system’s name and version here]
```

Reply:
Right now this only checks that you correctly include all of the lines, as if you were filling out a form, but I'm not sure that's the best approach. Some times issue templates follow a branching style, like if it's a bug fill out area A, a feature request area B, etc.

As I said before, I think a solid approach for step one would be checking that the issue body isn't exactly the issue template ( ie `!body.includes(issueTemplate)`)

Reply:
so this is what i thought about earlier, and `!body.includes(issueTemplate)` fails, if you look at this example:
```js
const body = ' Operating System Mac Version 1.7.6 Bug Description Wow  i have a bug!!!'
const issueTemplate = ' Operating System Version Bug Description'
```
`!body.includes(issueTemplate)` would return `false` here